tests/e2e/actor_reminder_partition/actor_reminder_partition_test.go

```diff
@@ -0,0 +1,168 @@
+	ActorID        string `json:"actorId,omitempty"`
+	StartTimestamp int    `json:"startTimestamp,omitempty"`
+	EndTimestamp   int    `json:"endTimestamp,omitempty"`
+}
+
+type actorReminder struct {
+	Data     string `json:"data,omitempty"`
+	DueTime  string `json:"dueTime,omitempty"`
+	Period   string `json:"period,omitempty"`
+	Callback string `json:"callback,omitempty"`
+}
+
+func parseLogEntries(resp []byte) []actorLogEntry {
+	logEntries := []actorLogEntry{}
+	err := json.Unmarshal(resp, &logEntries)
+	if err != nil {
+		return nil
+	}
+
+	return logEntries
+}
+
+func countActorAction(resp []byte, actorID string, action string) int {
+	count := 0
+	logEntries := parseLogEntries(resp)
+	for _, logEntry := range logEntries {
+		if (logEntry.ActorID == actorID) && (logEntry.Action == action) {
+			count++
+		}
+	}
+
+	return count
+}
+
+var tr *runner.TestRunner
+
+func TestMain(m *testing.M) {
+	// These apps will be deployed before starting actual test
+	// and will be cleaned up after all tests are finished automatically
+	testApps := []kube.AppDescription{
+		{
+			AppName:        appName,
+			DaprEnabled:    true,
+			ImageName:      "e2e-actorfeatures",
+			Replicas:       1,
+			IngressEnabled: true,
+			MetricsEnabled: true,
+			DaprCPULimit:   "2.0",
+			DaprCPURequest: "0.1",
+			AppCPULimit:    "2.0",
+			AppCPURequest:  "0.1",
+			AppEnv: map[string]string{
+				"TEST_APP_ACTOR_TYPE": "testactorreminderpartition",
+			},
+			Config: "actortypemetadata",
+		},
+	}
+
+	tr = runner.NewTestRunner(appName, testApps, nil, nil)
+	os.Exit(tr.Start(m))
+}
+
+func TestActorReminder(t *testing.T) {
+	externalURL := tr.Platform.AcquireAppExternalURL(appName)
+	require.NotEmpty(t, externalURL, "external URL must not be empty!")
+
+	logsURL := fmt.Sprintf(actorlogsURLFormat, externalURL)
+
+	// This initial probe makes the test wait a little bit longer when needed,
+	// making this test less flaky due to delays in the deployment.
+	t.Logf("Checking if app is healthy ...")
+	_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
+	require.NoError(t, err)
+
+	// Set reminder
+	reminder := actorReminder{
+		Data:    "reminderdata",
+		DueTime: "1s",
+		Period:  "1s",
+	}
+	reminderBody, err := json.Marshal(reminder)
+	require.NoError(t, err)
+
+	t.Run("Actor reminder changes number of partitions.", func(t *testing.T) {
+		for i := 0; i < numActorsPerThread; i++ {
+			if i == numActorsPerThread/2 {
+				// We change the number of partitions midway.
+				// This will test that the migration path also works.
+				// And validate that new reminders can work after increasing number of partitions.
+				tr.Platform.SetAppEnv(appName, "TEST_APP_ACTOR_REMINDERS_PARTITIONS", "5")
+			}
+
+			actorID := fmt.Sprintf(actorIDPartitionTemplate, i+1000)
+			// Deleting pre-existing reminder
+			_, err = utils.HTTPDelete(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName))
+			require.NoError(t, err)
```

Comment:
Wouldn't deleting the reminder negate the migration?

Reply:
No, deleting a reminder means updating the record with a list of reminders that removes one item from it. In the end, deleting all reminders will keep the list of reminder as empty after the migration - which is a valid scenario.